packages/base/Paginator/index.js

```diff
@@ -15,54 +16,99 @@ const paginationButtonSx = {
 };
 
 const Paginator = ({
+  fullPagintation,
   leftButtonContent,
   offset,
+  onPageClick,
   onNextClick,
   onPreviousClick,
   pageSize,
   pageTotal,
   rightButtonContent,
   total,
   ...rest
-}) => (
-  <Box
-    alignItems="center"
-    display="flex"
-    justifyContent="flex-end"
-    mt={5}
-    {...rest}
-  >
-    {pageTotal > 0 && (
-      <Box mr={3}>
-        {offset + 1}&ndash;{offset + pageTotal}
-        {total ? ` of ${total}` : ''}
-      </Box>
-    )}
-    <Button
-      disabled={offset === 0}
-      onClick={onPreviousClick}
-      sx={paginationButtonSx}
-    >
-      {leftButtonContent}
-    </Button>
-    <Button
-      disabled={
-        total
-          ? offset + pageSize >= total
-          : pageTotal === 0 || pageTotal % pageSize !== 0
-      }
-      onClick={onNextClick}
-      sx={paginationButtonSx}
+}) => {
+  const [currPage, setCurrPage] = useState(1);
+  const pagiBtns = [];
+
+  useEffect(() => {
+    setCurrPage(parseInt(offset / pageSize, 10) + 1);
+  }, [offset]);
```

Comment:
We don't need to keep track of this state. We can calculate it from offset and page size based on what's sent from the parent component.

```suggestion
  const currentPage = Math.floor(offset / pageSize) + 1;
```